assets/js/collaborative-editor/hooks/useWorkflow.ts

```diff
@@ -721,6 +722,17 @@ export const useCanRun = (): { canRun: boolean; tooltipMessage: string } => {
     isOldSnapshot,
   } = useWorkflowConditions();
 
+  // Get run limits from session context (defaults to allowed if missing)
+  const sessionContextStore = useContext(StoreContext)?.sessionContextStore;
+  const runLimits = sessionContextStore
+    ? useSyncExternalStore(
+        sessionContextStore.subscribe,
+        sessionContextStore.withSelector(
+          state => state.limits?.runs ?? { allowed: true, message: null }
+        )
+      )
+    : { allowed: true, message: null };
```

Comment:
According to the Rules Of Hooks in the React doc site [here](https://react.dev/warnings/invalid-hook-call-warning) it's a bad practice to call hooks in conditions. I am wondering if can't rewrite this like this:

```javascript
const sessionContextStore = useContext(StoreContext)?.sessionContextStore;
const runLimits = useSyncExternalStore(
  sessionContextStore?.subscribe ?? (() => () => {}),
  sessionContextStore?.withSelector(
    state => state.limits?.runs ?? { allowed: true, message: null }
  ) ?? (() => ({ allowed: true, message: null }))
);
```

What do you think ?